app/menus/NotificationMenu.tsx

```diff
@@ -0,0 +1,74 @@
+import { t } from "i18next";
+import { MoreIcon, SettingsIcon } from "outline-icons";
+import React from "react";
+import { MenuButton, useMenuState } from "reakit/Menu";
+import styled from "styled-components";
+import { s } from "@shared/styles";
+import ContextMenu from "~/components/ContextMenu";
+import Template from "~/components/ContextMenu/Template";
+import NudeButton from "~/components/NudeButton";
+import { actionToMenuItem, performAction } from "~/actions";
+import { navigateToNotificationSettings } from "~/actions/definitions/navigation";
+import { markNotificationsAsArchived } from "~/actions/definitions/notifications";
+import useActionContext from "~/hooks/useActionContext";
+import useOnClickOutside from "~/hooks/useOnClickOutside";
+import { hover } from "~/styles";
+import { MenuItem } from "~/types";
+
+const NotificationMenu: React.FC = () => {
+  const menuRef = React.useRef<HTMLDivElement>(null);
+  const menu = useMenuState();
+  const context = useActionContext();
+  const items: MenuItem[] = React.useMemo(
+    () => [
+      actionToMenuItem(markNotificationsAsArchived, context),
+      {
+        type: "button",
+        title: "Notification settings",
+        visible: true,
+        onClick: () => performAction(navigateToNotificationSettings, context),
+        icon: <SettingsIcon />,
+      },
+    ],
+    [context]
+  );
+
+  useOnClickOutside(
```

Comment:
I'm not sure this is achieving anything to be honest

Reply:
Without this, when the menu is open, clicking outside will close both the popup & the menu. Ideally, the user would have hoped to close just the menu.

Is this not working? 🤔

Reply:
That would be ideal, but it didn't seem to work when I tried

Reply:
That's weird, it worked when I tried now.

One case I've observed is, if we click inside the document, it closes all the popovers and menus. Maybe that's what is happening?

**Screencast for reference**
https://github.com/outline/outline/assets/132386067/16e986f8-4881-43f3-8f18-d749f381a3f9

Reply:
That must be it, it's ok for now